src/zcash/IncrementalMerkleTree.cpp

```diff
@@ -0,0 +1,287 @@
+#include <stdexcept>
+
+#include <boost/foreach.hpp>
+
+#include "zcash/IncrementalMerkleTree.hpp"
+#include "crypto/sha256.h"
+#include "zerocash/utils/util.h" // TODO: remove these utilities
+
+namespace libzcash {
+
+SHA256Compress SHA256Compress::combine(const SHA256Compress& a, const SHA256Compress& b)
```

Comment:
Oh I see, SHA256Compress represents an output, but it also effectively does define which hash function is being used by its implementation of `combine()`.